File: digest-util/src/test/java/cc/duduhuo/util/digest/TestBase64.java
/*
 * Copyright 2017-present Li Ying.
 * Licensed under the MIT License.
 */

package cc.duduhuo.util.digest;

import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;
import java.nio.charset.StandardCharsets;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertEquals;


public class TestBase64 {
    @Test
    public void test() throws IOException {
        assertEquals("YWJj", Base64.encodeToString("abc"));
        assertEquals(java.util.Base64.getEncoder().encodeToString("abc".getBytes(StandardCharsets.UTF_8)), Base64.encodeToString("abc"));
        assertEquals("YWJj", Base64.encodeToString("abc".getBytes()));
        String s = Base64.encodeToString(new File("for-testing-only.txt"));
        System.out.println("base64 = " + s);
        assertEquals(s, Base64.encodeToString(Base64.decodeToString(s)));
        assertArrayEquals(java.util.Base64.getDecoder().decode("YWJj".getBytes(StandardCharsets.UTF_8)), Base64.decode("YWJj"));
        Base64.decodeToFile(Base64.encode(new File("for-testing-only.txt")), new File("for-testing-only.txt2"));
    }
}

File: digest-util-ktx/src/test/kotlin/cc/duduhuo/util/digest/ktx/TestBase64.kt
/*
 * Copyright 2017-present Li Ying.
 * Licensed under the MIT License.
 */

package cc.duduhuo.util.digest.ktx

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.File
import java.io.IOException

class TestBase64 {
    @Test
    @Throws(IOException::class)
    fun test() {
        Assertions.assertEquals("YWJj", "abc".base64().encodeToString())
        Assertions.assertEquals("YWJj\n", "abc".base64().encodeToString(flags = cc.duduhuo.util.digest.Base64.DEFAULT))
        Assertions.assertEquals("YWJj", "abc".toByteArray().base64().encodeToString())
        val s = File("for-testing-only.txt").base64().encodeToString()
        println("base64 = $s")
        Assertions.assertEquals(s, s.base64().decodeToString().base64().encodeToString())
        File("for-testing-only.txt").base64().encode().base64().decodeToFile(out = File("for-testing-only.txt2"))
    }
}
